cmd/kubeadm/app/cmd/config.go

```diff
@@ -143,7 +143,12 @@ func getDefaultAPIObjectBytes(apiObject string) ([]byte, error) {
 	if err != nil {
 		return []byte{}, err
 	}
-	return kubeadmutil.MarshalToYamlForCodecs(internalcfg, kubeadmapiv1alpha3.SchemeGroupVersion, kubeadmscheme.Codecs)
+	switch internalcfg.(type) {
+	case *kubeadmapi.MasterConfiguration:
+		return configutil.MarshalMasterConfigurationToBytes(internalcfg.(*kubeadmapi.MasterConfiguration), kubeadmapiv1alpha3.SchemeGroupVersion, true)
```

Comment:
I appreciate that we strive for cleanness and we skip printing component config if equal to default.
However I think that we should add a flag to allow users to ask explicitly to print also component config defaults

Reply:
Yes, this is coming later in the `kubeadm config print-defaults` command.